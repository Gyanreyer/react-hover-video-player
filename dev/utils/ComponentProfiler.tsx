/* eslint-disable no-console */
import React from 'react';

interface RenderTiming {
  averageRenderTime: number;
  renderCount: number;
}

interface ComponentProfilerProps {
  profilerID: string;
  children: React.ReactChild;
}

/**
 * Util component logs out render timings for whatever component it's wrapped aroud
 */
const ComponentProfiler = ({
  profilerID,
  children,
}: ComponentProfilerProps): JSX.Element => {
  // Use a ref to track render timing for this component in a persistent store
  const renderTiming = React.useRef<RenderTiming>();
  if (!renderTiming.current) {
    renderTiming.current = {
      averageRenderTime: 0,
      renderCount: 0,
    };
  }

  // Logs out helpful render timing info for performance measurements
  const onProfilerRender = React.useCallback(
    (
      id, // the "id" prop of the Profiler tree that has just committed
      phase: string, // either "mount" (if the tree just mounted) or "update" (if it re-rendered)
      actualDuration: number // time spent rendering the committed update
    ) => {
      if (phase === 'mount') {
        console.log(`${profilerID} | MOUNT: ${actualDuration}ms`);
      } else {
        renderTiming.current.renderCount += 1;
        renderTiming.current.averageRenderTime +=
          (actualDuration - renderTiming.current.averageRenderTime) /
          renderTiming.current.renderCount;
        console.log(
          `${profilerID} | UPDATE: ${actualDuration}ms | New average: ${renderTiming.current.averageRenderTime}ms`
        );
      }
    },
    [profilerID]
  );
  return (
    <React.Profiler id={profilerID} onRender={onProfilerRender}>
      {children}
    </React.Profiler>
  );
};

export default ComponentProfiler;
