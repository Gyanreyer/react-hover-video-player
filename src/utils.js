/**
 * @typedef   {object}  VideoSource
 * @property  {string}  src - The src URL string to use for a video player source
 * @property  {string}  type - The media type of the video, ie 'video/mp4'
 */

/**
 * @function  formatVideoSrc
 *
 * Takes a videoSrc value and formats it as an array of VideoSource objects which can be used to render
 * <source> elements for the video
 *
 * @param {!(string|string[]|VideoSource|VideoSource[])}  videoSrc - Source(s) to use for the video player. Accepts 3 different formats:
 *                                                                   - **String**: the URL string to use as the video player's src
 *                                                                   - **Object**: an object with attributes:
 *                                                                     - src: The src URL string to use for a video player source
 *                                                                     - type: The media type of the video source, ie 'video/mp4'
 *                                                                   - **Array**: if you would like to provide multiple sources, you can provide an array of URL strings and/or objects with the shape described above
 *
 * @returns {VideoSource[]} Array of formatted VideoSource objects which can be used to render <source> elements for the video
 */
export function formatVideoSrc(videoSrc) {
  if (videoSrc == null) {
    // A videoSrc value is required in order to make the video player work
    console.error(
      "Error: 'videoSrc' prop is required for HoverVideoPlayer component"
    );

    return [];
  }

  return (
    // Make sure we can treat the videoSrc value as an array
    []
      .concat(videoSrc)
      // Parse our video source values into an array of VideoSource objects that can be used to render sources for the video
      .reduce((sourceArray, source) => {
        if (typeof source === 'string') {
          // If the source is a string, it's an src URL so format it into a VideoSource object and add it to the array
          sourceArray.push({ src: source });
        } else if (source && source.src) {
          // If the source is an object with an src, just add it to the array
          sourceArray.push({ src: source.src, type: source.type });
        } else {
          // Log an error if one of the videoSrc values is invalid
          console.error(
            "Error: invalid value provided to HoverVideoPlayer prop 'videoSrc':",
            source
          );
        }

        return sourceArray;
      }, [])
  );
}

/**
 * @typedef   {object}  VideoCaptionsTrack
 * @property  {string}  src - The src URL string for the captions track file
 * @property  {string}  srcLang - The language code for the language that these captions are in
 * @property  {string}  label - The title of the captions track
 * @property  {bool}    default - Whether this track should be used by default if the user's preferences don't match an available srcLang
 */

/**
 * @function formatVideoCaptions
 *
 * Takes a videoCaptions value and formats it as an array of VideoCaptionsTrack objects which can be used to render
 * <track> elements for the video
 *
 * @param {!(string|string[]|VideoCaptionsTrack|VideoCaptionsTrack[])} videoCaptions - Captions track(s) to use for the video player for accessibility. Accepts 3 different formats:
 *                                                                                     - **String**: the URL string to use as the captions track's src
 *                                                                                     - **Object**: an object with attributes:
 *                                                                                       - src: The src URL string for the captions track file
 *                                                                                       - srcLang: The language code for the language that these captions are in (ie, 'en', 'es', 'fr')
 *                                                                                       - label: The title of the captions track
 *                                                                                       - default: Whether this track should be used by default if the user's preferences don't match an available srcLang
 *                                                                                     - **Array**: if you would like to provide multiple caption tracks, you can provide an array of objects with the shape described above
 *
 * @returns {VideoCaptionsTrack[]}  Array of formatted VideoCaptionsTrack objects which can be used to render <track> elements for the video
 */
export function formatVideoCaptions(videoCaptions) {
  // If no captions were provided, return an empty array
  if (videoCaptions == null) return [];

  return (
    // Make sure we can treat the videoSrc value as an array
    []
      .concat(videoCaptions)
      // Parse our video captions values into an array of VideoCaptionsTrack
      // objects that can be used to render caption tracks for the video
      .reduce((captionsArray, captions) => {
        if (captions && captions.src) {
          captionsArray.push({
            src: captions.src,
            srcLang: captions.srcLang,
            label: captions.label,
            default: Boolean(captions.default),
          });
        } else {
          // Log an error if one of the videoCaptions values is invalid
          console.error(
            "Error: invalid value provided to HoverVideoPlayer prop 'videoCaptions'",
            captions
          );
        }

        return captionsArray;
      }, [])
  );
}
