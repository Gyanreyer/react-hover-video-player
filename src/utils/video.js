/* eslint-disable no-param-reassign */
// Enumerates states that the video can be in
export const VIDEO_STATE = {
  paused: 'paused',
  loading: 'loading',
  playing: 'playing',
};

/**
 * @function getVideoState
 *
 * Takes a video element and returns its current playing state
 *
 * @param {node} videoElement
 */
export function getVideoState(videoElement) {
  if (videoElement.paused || videoElement.ended) {
    return VIDEO_STATE.paused;
  }

  // If the video isn't paused but its readyState indicates it isn't loaded enough
  // to play yet, it is loading
  if (videoElement.readyState < 3) {
    return VIDEO_STATE.loading;
  }

  // If the video isn't paused and its ready state indicates it's loaded enough to play,
  // assume it's playing
  return VIDEO_STATE.playing;
}

/**
 * @function playVideo
 *
 * Takes a video element and attempts to play it.
 *
 * @param {node}  videoElement - The video element to start playing
 *
 * @returns {Promise}   A Promise which will resolve when the video starts playing or reject if an error occurs
 */
export function playVideo(videoElement) {
  // Start playing the video and hang onto the play promise it returns
  let playPromise = videoElement.play();

  if (!playPromise || !playPromise.then) {
    // If videoElement.play() didn't return a promise, we'll manually create one
    // ourselves which mimics the same behavior
    playPromise = new Promise((resolve, reject) => {
      // Declaring onVideoPlaybackFailed up here so we can refer to it and remove its event listener
      // if the video successfully starts playing
      let onVideoPlaybackFailed;

      // Set up event listener to resolve the promise when the video player starts playing
      const onVideoPlaybackStarted = () => {
        // Remove the event listeners we added as cleanup now that the play attempt has succeeded
        videoElement.removeEventListener('playing', onVideoPlaybackStarted);
        videoElement.removeEventListener('error', onVideoPlaybackFailed);

        // Resolve because we successfully started playing!
        resolve();
      };
      videoElement.addEventListener('playing', onVideoPlaybackStarted, {
        once: true,
      });

      // Set up event listener to reject the promise when the video player encounters an error
      onVideoPlaybackFailed = (event) => {
        // Remove the event listeners we added as cleanup now that the play attempt has failed
        videoElement.removeEventListener('error', onVideoPlaybackFailed);
        videoElement.removeEventListener('playing', onVideoPlaybackStarted);

        // Reject with the error that was thrown
        reject(event.error);
      };
      videoElement.addEventListener('error', onVideoPlaybackFailed);
    });
  }

  return playPromise;
}

/**
 * @function  pauseVideo
 *
 * Pauses the video and resets it to the beginning if necessary
 *
 * @param {node}  videoElement - The video element to pause playback for
 * @param {bool}  shouldRestart - Whether the videp should be reset to the beginning after pausing
 */
export function pauseVideo(videoElement, shouldRestart) {
  videoElement.pause();

  if (shouldRestart) {
    // If we should restart the video, reset its time to the beginning
    videoElement.currentTime = 0;
  }
}
