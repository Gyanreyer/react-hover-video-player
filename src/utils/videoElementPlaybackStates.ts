/**
 * Takes a video element and returns whether it is currently paused, meaning it is not
 * playing or attempting to play.
 *
 * @param {HTMLVideoElement} videoElement
 */
export function isVideoElementPaused(
  videoElement: HTMLVideoElement | null
): boolean {
  return videoElement ? videoElement.paused || videoElement.ended : true;
}
