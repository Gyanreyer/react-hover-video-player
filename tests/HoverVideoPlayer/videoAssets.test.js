import { renderHoverVideoPlayer } from '../utils';

describe('videoSrc prop', () => {
  describe('Handles valid videoSrc prop values correctly', () => {
    test('correctly handles receiving a string for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: '/fake/video-file.mp4',
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(1);
      expect(videoSources[0]).toHaveAttribute('src', '/fake/video-file.mp4');
      expect(videoSources[0]).not.toHaveAttribute('type');
    });

    test('correctly handles receiving an array of strings for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: ['/fake/video-file.webm', '/fake/video-file.mp4'],
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(2);
      expect(videoSources[0]).toHaveAttribute('src', '/fake/video-file.webm');
      expect(videoSources[0]).not.toHaveAttribute('type');
      expect(videoSources[1]).toHaveAttribute('src', '/fake/video-file.mp4');
      expect(videoSources[1]).not.toHaveAttribute('type');
    });

    test('correctly handles receiving a valid object for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: { src: '/fake/video-file.mp4', type: 'video/mp4' },
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(1);
      expect(videoSources[0]).toHaveAttribute('src', '/fake/video-file.mp4');
      expect(videoSources[0]).toHaveAttribute('type', 'video/mp4');
    });

    test('correctly handles receiving an array of objects for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: [
          { src: '/fake/video-file.webm', type: 'video/webm' },
          { src: '/fake/video-file.mp4', type: 'video/mp4' },
        ],
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(2);
      expect(videoSources[0]).toHaveAttribute('src', '/fake/video-file.webm');
      expect(videoSources[0]).toHaveAttribute('type', 'video/webm');
      expect(videoSources[1]).toHaveAttribute('src', '/fake/video-file.mp4');
      expect(videoSources[1]).toHaveAttribute('type', 'video/mp4');
    });

    test('correctly handles receiving an array with a mix of strings and objects for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: [
          '/fake/video-file.webm',
          { src: '/fake/video-file.avi' },
          { src: '/fake/video-file.mp4', type: 'video/mp4' },
        ],
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(3);
      expect(videoSources[0]).toHaveAttribute('src', '/fake/video-file.webm');
      expect(videoSources[0]).not.toHaveAttribute('type');
      expect(videoSources[1]).toHaveAttribute('src', '/fake/video-file.avi');
      expect(videoSources[1]).not.toHaveAttribute('type');
      expect(videoSources[2]).toHaveAttribute('src', '/fake/video-file.mp4');
      expect(videoSources[2]).toHaveAttribute('type', 'video/mp4');
    });
  });

  describe('Handles invalid videoSrc prop values correctly', () => {
    let originalConsoleError;

    beforeEach(() => {
      // Mock the console.error function so we can verify that an error was logged correctly
      originalConsoleError = console.error;
      console.error = jest.fn();
    });

    afterEach(() => {
      console.error = originalConsoleError;
    });

    test('correctly handles not receiving a videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({});

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(0);

      // Should have logged an error warning that the 'videoSrc' prop is required
      expect(console.error).toHaveBeenCalledTimes(1);
      expect(console.error).toHaveBeenCalledWith(
        "Error: 'videoSrc' prop is required for HoverVideoPlayer component"
      );
    });

    test('correctly handles receiving a single invalid value for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: 100,
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(0);

      // Should have logged an error warning that the 'videoSrc' prop is required
      expect(console.error).toHaveBeenCalledTimes(1);
      expect(console.error).toHaveBeenCalledWith(
        "Error: invalid value provided to HoverVideoPlayer prop 'videoSrc':",
        100
      );
    });

    test('correctly handles receiving an invalid value in an array for the videoSrc prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: [
          'valid-video-file.webm',
          false,
          { src: 'valid-video-file.mp4', type: 'video/mp4' },
        ],
      });

      const videoElement = container.querySelector('video');

      const videoSources = videoElement.querySelectorAll('source');
      expect(videoSources).toHaveLength(2);
      expect(videoSources[0]).toHaveAttribute('src', 'valid-video-file.webm');
      expect(videoSources[0]).not.toHaveAttribute('type');
      expect(videoSources[1]).toHaveAttribute('src', 'valid-video-file.mp4');
      expect(videoSources[1]).toHaveAttribute('type', 'video/mp4');

      // Should have logged an error warning that the 'videoSrc' prop is required
      expect(console.error).toHaveBeenCalledTimes(1);
      expect(console.error).toHaveBeenCalledWith(
        "Error: invalid value provided to HoverVideoPlayer prop 'videoSrc':",
        false
      );
    });
  });
});

describe('videoCaptions prop', () => {
  describe('Handles valid videoCaptions prop values correctly', () => {
    test('correctly handles receiving a string for the videoCaptions prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: '/fake/video-file.mp4',
        videoCaptions: '/fake/captions-file.vtt',
      });

      const videoElement = container.querySelector('video');

      const videoTracks = videoElement.querySelectorAll('track');
      expect(videoTracks).toHaveLength(1);
      expect(videoTracks[0]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[0]).toHaveAttribute('src', '/fake/captions-file.vtt');
    });

    test('correctly handles receiving a valid object for the videoCaptions prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: '/fake/video-file.mp4',
        videoCaptions: {
          src: '/fake/captions-file-en.vtt',
          srcLang: 'en',
          label: 'English',
        },
      });

      const videoElement = container.querySelector('video');

      const videoTracks = videoElement.querySelectorAll('track');
      expect(videoTracks).toHaveLength(1);
      expect(videoTracks[0]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[0]).toHaveAttribute(
        'src',
        '/fake/captions-file-en.vtt'
      );
      expect(videoTracks[0]).toHaveAttribute('srclang', 'en');
      expect(videoTracks[0]).toHaveAttribute('label', 'English');
    });

    test('correctly handles receiving an array of objects for the videoCaptions prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: '/fake/video-file.mp4',
        videoCaptions: [
          {
            src: '/fake/captions-file-en.vtt',
            srcLang: 'en',
            label: 'English',
          },
          {
            src: '/fake/captions-file-fr.vtt',
            srcLang: 'fr',
            label: 'French',
          },
          {
            src: '/fake/captions-file-de.vtt',
            srcLang: 'de',
            label: 'German',
          },
        ],
      });

      const videoElement = container.querySelector('video');

      const videoTracks = videoElement.querySelectorAll('track');
      expect(videoTracks).toHaveLength(3);
      expect(videoTracks[0]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[0]).toHaveAttribute(
        'src',
        '/fake/captions-file-en.vtt'
      );
      expect(videoTracks[0]).toHaveAttribute('srclang', 'en');
      expect(videoTracks[0]).toHaveAttribute('label', 'English');

      expect(videoTracks[1]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[1]).toHaveAttribute(
        'src',
        '/fake/captions-file-fr.vtt'
      );
      expect(videoTracks[1]).toHaveAttribute('srclang', 'fr');
      expect(videoTracks[1]).toHaveAttribute('label', 'French');

      expect(videoTracks[2]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[2]).toHaveAttribute(
        'src',
        '/fake/captions-file-de.vtt'
      );
      expect(videoTracks[2]).toHaveAttribute('srclang', 'de');
      expect(videoTracks[2]).toHaveAttribute('label', 'German');
    });
  });

  describe('Handles invalid videoCaptions prop values correctly', () => {
    let originalConsoleError;

    beforeEach(() => {
      // Mock the console.error function so we can verify that an error was logged correctly
      originalConsoleError = console.error;
      console.error = jest.fn();
    });

    afterEach(() => {
      console.error = originalConsoleError;
    });

    test('correctly handles receiving a single invalid value for the videoCaptions prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: 'fake/video-file.mp4',
        videoCaptions: false,
      });

      const videoElement = container.querySelector('video');

      const videoTracks = videoElement.querySelectorAll('track');
      expect(videoTracks).toHaveLength(0);

      // Should have logged an error warning that the 'videoSrc' prop is required
      expect(console.error).toHaveBeenCalledTimes(1);
      expect(console.error).toHaveBeenCalledWith(
        "Error: invalid value provided to HoverVideoPlayer prop 'videoCaptions'",
        false
      );
    });

    test('correctly handles receiving an invalid value in an array for the videoCaptions prop', () => {
      const { container } = renderHoverVideoPlayer({
        videoSrc: 'fake/video-file.mp4',
        videoCaptions: [
          {
            src: '/fake/captions-file-en.vtt',
            srcLang: 'en',
            label: 'English',
          },
          false,
          100,
          {
            src: '/fake/captions-file-fr.vtt',
            srcLang: 'fr',
            label: 'French',
          },
        ],
      });

      const videoElement = container.querySelector('video');

      const videoTracks = videoElement.querySelectorAll('track');
      expect(videoTracks).toHaveLength(2);
      expect(videoTracks[0]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[0]).toHaveAttribute(
        'src',
        '/fake/captions-file-en.vtt'
      );
      expect(videoTracks[0]).toHaveAttribute('srclang', 'en');
      expect(videoTracks[0]).toHaveAttribute('label', 'English');

      expect(videoTracks[1]).toHaveAttribute('kind', 'captions');
      expect(videoTracks[1]).toHaveAttribute(
        'src',
        '/fake/captions-file-fr.vtt'
      );
      expect(videoTracks[1]).toHaveAttribute('srclang', 'fr');
      expect(videoTracks[1]).toHaveAttribute('label', 'French');

      // Should have logged an error warning that the 'videoSrc' prop is required
      expect(console.error).toHaveBeenCalledTimes(2);
      expect(console.error).toHaveBeenNthCalledWith(
        1,
        "Error: invalid value provided to HoverVideoPlayer prop 'videoCaptions'",
        false
      );
      expect(console.error).toHaveBeenNthCalledWith(
        2,
        "Error: invalid value provided to HoverVideoPlayer prop 'videoCaptions'",
        100
      );
    });
  });
});
