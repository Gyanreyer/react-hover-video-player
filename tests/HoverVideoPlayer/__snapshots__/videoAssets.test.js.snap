// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`videoCaptions prop Handles invalid videoCaptions prop values correctly correctly handles receiving a single invalid value for the videoCaptions prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="fake/video-file.mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoCaptions prop Handles invalid videoCaptions prop values correctly correctly handles receiving an invalid value in an array for the videoCaptions prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="fake/video-file.mp4"
      />
      <track
        kind="captions"
        label="English"
        src="/fake/captions-file-en.vtt"
        srclang="en"
      />
      <track
        kind="captions"
        label="French"
        src="/fake/captions-file-fr.vtt"
        srclang="fr"
      />
    </video>
  </div>
</div>
`;

exports[`videoCaptions prop Handles valid videoCaptions prop values correctly correctly handles receiving a string for the videoCaptions prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.mp4"
      />
      <track
        kind="captions"
        src="/fake/captions-file.vtt"
      />
    </video>
  </div>
</div>
`;

exports[`videoCaptions prop Handles valid videoCaptions prop values correctly correctly handles receiving a valid object for the videoCaptions prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.mp4"
      />
      <track
        kind="captions"
        label="English"
        src="/fake/captions-file-en.vtt"
        srclang="en"
      />
    </video>
  </div>
</div>
`;

exports[`videoCaptions prop Handles valid videoCaptions prop values correctly correctly handles receiving an array of objects for the videoCaptions prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.mp4"
      />
      <track
        kind="captions"
        label="English"
        src="/fake/captions-file-en.vtt"
        srclang="en"
      />
      <track
        kind="captions"
        label="French"
        src="/fake/captions-file-fr.vtt"
        srclang="fr"
      />
      <track
        kind="captions"
        label="German"
        src="/fake/captions-file-de.vtt"
        srclang="de"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles invalid videoSrc prop values correctly correctly handles not receiving a videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    />
  </div>
</div>
`;

exports[`videoSrc prop Handles invalid videoSrc prop values correctly correctly handles receiving a single invalid value for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    />
  </div>
</div>
`;

exports[`videoSrc prop Handles invalid videoSrc prop values correctly correctly handles receiving an invalid value in an array for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="valid-video-file.webm"
      />
      <source
        src="valid-video-file.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles valid videoSrc prop values correctly correctly handles receiving a string for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles valid videoSrc prop values correctly correctly handles receiving a valid object for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles valid videoSrc prop values correctly correctly handles receiving an array of objects for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.webm"
        type="video/webm"
      />
      <source
        src="/fake/video-file.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles valid videoSrc prop values correctly correctly handles receiving an array of strings for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.webm"
      />
      <source
        src="/fake/video-file.mp4"
      />
    </video>
  </div>
</div>
`;

exports[`videoSrc prop Handles valid videoSrc prop values correctly correctly handles receiving an array with a mix of strings and objects for the videoSrc prop 1`] = `
.emotion-1 {
  position: relative;
}

.emotion-0 {
  width: 100%;
  display: block;
  object-fit: cover;
}

<div>
  <div
    class="emotion-1"
    data-testid="hover-video-player-container"
  >
    <video
      class="emotion-0"
      loop=""
      playsinline=""
      preload="metadata"
    >
      <source
        src="/fake/video-file.webm"
      />
      <source
        src="/fake/video-file.avi"
      />
      <source
        src="/fake/video-file.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;
